trace/tracestate.go

```diff
@@ -260,6 +260,17 @@ func (ts TraceState) Get(key string) string {
 	return ""
 }
 
+// Keys return all the keys from the TraceState in the insertion order.
+func (ts TraceState) Keys() []string {
```

Comment:
This modifies the stable public API. So if we do merge this, we won't be able to modify this method anymore.
As such, this change should have its own changelog entry.

Also, I wonder if the sorting is really necessary (or should it happen within folk's code)

Reply:
- Removed the reverse logic
- Added the CHANGE log. 
- Spec didn't require to expose key list. It only specify the minimum API needed. In Java, it exposes [asMap()](https://github.com/open-telemetry/opentelemetry-java/blob/3a597446b34d212e55a110d6f288586fed35a7de/api/all/src/main/java/io/opentelemetry/api/trace/TraceState.java#L76). In CPP, it provides [GetAllEntries](https://github.com/open-telemetry/opentelemetry-cpp/blob/84368cd76151bfad525739d675061bd071477f81/api/include/opentelemetry/trace/trace_state.h#L198)

Reply:
Adding `Entries` instead of `Keys` would be more useful (you would not need to get the values afterwards).

However, notice that in Java there is also a [foreach()](https://github.com/open-telemetry/opentelemetry-java/blob/3a597446b34d212e55a110d6f288586fed35a7de/api/all/src/main/java/io/opentelemetry/api/trace/TraceState.java#L72-L73). I like this idea, because it can reduce the heap allocations (no need to allocate the slice that would be returned). I propose to add a `func (ts TraceState) Walk(f func(key, value string) bool)` instead. See here a similar implementation: https://github.com/open-telemetry/opentelemetry-go/blob/log/v0.4.0/log/record.go#L96-L109

Reply:
I was considering to expose members directly. However as member is a private struct. To make it public will again cause bigger API changes. If convert it to tuples, it will create more fragmented objects in heap. Let me add `foreach()` instead